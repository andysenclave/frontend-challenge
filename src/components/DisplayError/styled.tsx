import styled from 'styled-components';

export const ErrorMessage = styled.p`
  font-size: 1.2em;
  background: pink;
  color: red;
`;