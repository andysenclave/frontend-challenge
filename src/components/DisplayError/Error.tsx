import React from 'react';

import { ErrorMessage } from './styled';

interface ErrorProps {
  message: string;
}

const displayError = ({ message }: ErrorProps) => (
  <ErrorMessage>
    {message}
  </ErrorMessage>
);

export default displayError;
