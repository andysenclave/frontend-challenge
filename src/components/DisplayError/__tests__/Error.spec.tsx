import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import DisplayError from '../Error';

describe('test Error component', () => {
  it('renders the correct error message', () => {
    const errorMessage = 'Test message';
    const errorComponent = shallow(<DisplayError message={errorMessage}/>);
    
    expect(errorComponent.text()).toEqual(errorMessage);
  });

  it('matches the snapshot', () => {
    const errorMessage = 'Test message';
    const tree = renderer.create(<DisplayError message={errorMessage}/>).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('color', 'red');
  });
});