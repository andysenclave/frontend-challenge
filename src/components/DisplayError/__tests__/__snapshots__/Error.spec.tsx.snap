// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Error component matches the snapshot 1`] = `
.c0 {
  font-size: 1.2em;
  background: pink;
  color: red;
}

<p
  className="c0"
>
  Test message
</p>
`;
